src/mixings/logit_sb_mixing.cc

```diff
@@ -0,0 +1,154 @@
+    Eigen::VectorXd prior_vec =
+        bayesmix::to_eigen(priorcast->normal_prior().mean());
+    dim = prior_vec.size();
+    if (dim != precision.cols()) {
+      throw std::invalid_argument(
+          "Hyperparameters dimensions are not consisent");
+    }
+    if (priorcast->proposal_var() <= 0) {
+      throw std::invalid_argument("Proposal variance parameter must be > 0");
+    }
+    if (priorcast->step_size() <= 0) {
+      throw std::invalid_argument("Step size parameter must be > 0");
+    }
+    state.regression_coeffs = Eigen::MatrixXd(dim, num_clusters);
+    for (int i = 0; i < num_clusters; i++) {
+      state.regression_coeffs.col(i) = prior_vec;
+    }
+
+  } else {
+    throw std::invalid_argument("Unrecognized mixing prior");
+  }
+}
+
+Eigen::VectorXd LogitSBMixing::grad_log_full_cond(
+    const Eigen::VectorXd &alpha, const std::vector<bool> &is_curr_clus,
+    const std::vector<bool> &is_prev_clus) {
+  auto priorcast = cast_prior();
+  Eigen::VectorXd prior_mean =
+      bayesmix::to_eigen(priorcast->normal_prior().mean());
+  Eigen::VectorXd grad = precision * (prior_mean - alpha);
+  for (int i = 0; i < is_curr_clus.size(); i++) {
+    double prod = covariates_ptr->row(i).dot(alpha);
+    grad += -(is_curr_clus[i] + (1.0 - is_prev_clus[i]) * sigmoid(-prod)) *
+            covariates_ptr->row(i);
+  }
+  return grad;
+}
+
+double LogitSBMixing::log_like(const Eigen::VectorXd &alpha,
+                               const std::vector<bool> &is_curr_clus,
+                               const std::vector<bool> &is_prev_clus) {
+  double like = 0.0;
+  for (int i = 0; i < is_curr_clus.size(); i++) {
+    double prod = covariates_ptr->row(i).dot(alpha);
+    like += -is_curr_clus[i] * prod + (1.0 - is_prev_clus[i]) * sigmoid(prod);
+  }
+  return like;
+}
+
+void LogitSBMixing::update_state(
+    const std::vector<std::shared_ptr<AbstractHierarchy>> &unique_values,
+    const std::vector<unsigned int> &allocations, const unsigned int n) {
+  // Langevin-Adjusted Metropolis-Hastings step
+  auto &rng = bayesmix::Rng::Instance().get();
+  Eigen::VectorXd state_c = state.regression_coeffs;
+  auto priorcast = cast_prior();
+  Eigen::VectorXd prior_mean =
+      bayesmix::to_eigen(priorcast->normal_prior().mean());
+  double prop_var = priorcast->proposal_var();
+  double step = priorcast->step_size();
+  // Loop over clusters
+  for (int h = 0; h < unique_values.size(); h++) {
+    // Compute allocation indicators
+    std::vector<bool> is_curr_clus(n);
+    std::vector<bool> is_prev_clus(n);
+    for (int i = 0; i < n; i++) {
+      if (allocations[i] == h) {
+        is_curr_clus[i] = true;
+      } else if (allocations[i] < h) {
+        is_prev_clus[i] = true;
+      }
+    }
+    // Draw proposed state from its distribution
+    Eigen::VectorXd prop_mean =
+        state_c +
+        step * grad_log_full_cond(state_c, is_curr_clus, is_prev_clus);
+    auto prop_covar = prop_var * Eigen::MatrixXd::Identity(dim, dim);
```

Comment:
This is not a proper MALA step, since in MALA the step size controls also the covariance of the proposal

Reply:
Right, you have to use the square root just like you did in your snippet, I always forget about it.